src/typescript/frontend/src/lib/aptos-indexer/fungible-assets.ts

```diff
@@ -115,3 +110,15 @@ export async function fetchAllFungibleAssetsBalance({
     return address.emojicoin.toString() === parseTypeTag(token.asset_type).toString();
   });
 }
+
+async function fetchHoldersInternal(coinAddress: string) {
+  const holders = await fetchFungibleAssetsBalance({ assetType: coinAddress });
+  // Exclude the factory address from the holders list.
+  return holders.current_fungible_asset_balances.filter(
+    (h) => h.owner_address !== coinAddress.replace("::coin_factory::Emojicoin", "")
+  );
+}
+
+export const fetchHolders = unstable_cache(fetchHoldersInternal, ["fetch-holders"], {
```

Comment:
```suggestion
export const fetchCachedHolders = unstable_cache(fetchHoldersInternal, ["fetch-holders"], {
```

Just to make it clear it's cached when calling it